src/planner/plan/plan_port_db.cpp

```diff
@@ -27,22 +27,46 @@ std::unique_ptr<LogicalPlan> Planner::planExportDatabase(const BoundStatement& s
     auto plan = std::make_unique<LogicalPlan>();
     auto fileTypeStr = FileTypeUtils::toString(fileType);
     StringUtils::toLower(fileTypeStr);
-    auto copyToSuffix = "." + fileTypeStr;
-    std::string name = common::stringFormat("COPY_{}", FileTypeUtils::toString(fileType));
-    auto entry =
-        clientContext->getCatalog()->getFunctionEntry(clientContext->getTransaction(), name);
-    auto func = function::BuiltInFunctionsUtils::matchFunction(name,
-        entry->ptrCast<FunctionCatalogEntry>());
-    KU_ASSERT(func != nullptr);
-    auto exportFunc = *func->constPtrCast<function::ExportFunction>();
+    auto userFileSuffix = "." + fileTypeStr;
+    auto internalFileSuffix = "." + StringUtils::getLower(FileTypeUtils::toString(FileType::CSV));
+    auto userTableCopyToFuncName =
+        common::stringFormat("COPY_{}", FileTypeUtils::toString(fileType));
+    auto internalTableCopyToFuncName =
+        common::stringFormat("COPY_{}", FileTypeUtils::toString(FileType::CSV));
+    auto userTableCopyFunc = function::BuiltInFunctionsUtils::matchFunction(userTableCopyToFuncName,
+        clientContext->getCatalog()
+            ->getFunctionEntry(clientContext->getTransaction(), userTableCopyToFuncName)
+            ->ptrCast<FunctionCatalogEntry>());
+    auto internalTableCopyFunc =
+        function::BuiltInFunctionsUtils::matchFunction(internalTableCopyToFuncName,
+            clientContext->getCatalog()
+                ->getFunctionEntry(clientContext->getTransaction(), internalTableCopyToFuncName)
+                ->ptrCast<FunctionCatalogEntry>());
+    KU_ASSERT(userTableCopyFunc != nullptr && internalTableCopyFunc != nullptr);
+    auto userTableExportFunc = userTableCopyFunc->constPtrCast<function::ExportFunction>();
+    auto internalTableExportFunc = internalTableCopyFunc->constPtrCast<function::ExportFunction>();
```

Comment:
Seems we should resolve this at binding stage? Also the logic seems very similar so you might wanna refactor a bit.

Reply:
I already removed this change in my latest commit